src/main/java/core/mate/academy/service/impl/TruckProducerImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.mate.academy.service.impl;
+
+import core.mate.academy.model.Truck;
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducerImpl implements MachineProducer<Truck> {
+    private List<Truck> trucks;
```

Comment:
The visibility of the 'trucks' field should be private to adhere to encapsulation principles. Exposing fields directly is not a recommended practice.